tools/substitute.py

```diff
@@ -0,0 +1,29 @@
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+import argparse
+import os
+import os.path
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--input-file")
+    parser.add_argument("--output-file")
+    parser.add_argument("--install_dir")
+    parser.add_argument("--replace", action="append", nargs=2)
+    options = parser.parse_args()
+
+    with open(options.input_file, "r") as f:
+        contents = f.read()
+
+    output_file = os.path.join(options.install_dir, options.output_file)
+    try:
+        os.makedirs(os.path.dirname(output_file))
+    except OSError:
+        pass
```

Comment:
Wouldn't `exists_ok` be the right solution here (again, present only in Python-3.2+)
```suggestion
     os.makedirs(os.path.dirname(output_file), exists_ok=True)
```

Reply:
I just copied this file. I will do what you said and see if it works.